docs/internals/generic/_index.md

```diff
@@ -0,0 +1,215 @@
+
+{{% snippet "show-secret.md" %}}
+
+When executed, this sample code will create a connection to your machine as a client.
+Then control your machine programmatically by getting your `generic` service from the machine with `FromRobot` and adding API method calls, as shown in the following examples.
+
+Be sure to import the generic service package for the SDK you are using:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+```python
+from viam.services.generic import Generic
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```go
+import (
+  "go.viam.com/rdk/services/generic"
+)
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+```cpp
+#include <viam/sdk/services/generic/generic.hpp>
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## API
+
+The generic service supports the following method:
+
+{{< readfile "/static/include/services/apis/generic.md" >}}
+
+### DoCommand
+
+Execute model-specific commands.
+If you are implementing your own generic service and add features that have no built-in API method, you can access them with `DoCommand`.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `command` [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): The command to execute.
+
+**Returns:**
+
+- [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): Result of the executed command.
+
+```python {class="line-numbers linkable-line-numbers"}
+my_generic = Generic.from_robot(robot=robot, name="my_generic_service")
+
+raw_dict = {
+  "command": "raw",
+  "raw_input": "home"
+}
+await my_generic.do_command(raw_dict)
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/generic/client/index.html#viam.services.generic.client.GenericClient.do_command).
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cmd` [(map[string]interface{})](https://go.dev/blog/maps): The command to execute.
+
+**Returns:**
+
+- [(map[string]interface{})](https://go.dev/blog/maps): Result of the executed command.
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go {class="line-numbers linkable-line-numbers"}
+myGeneric, err := generic.FromRobot(robot, "my_generic_service")
+
+resp, err := myGeneric.DoCommand(ctx, map[string]interface{}{"command": "example"})
+```
+
+For more information, see the [Go SDK Code](https://github.com/viamrobotics/api/blob/main/service/generic/v1/generic_grpc.pb.go).
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+**Parameters:**
+
+- `command` [(string)](https://cplusplus.com/reference/string/string/): The command to execute.
+
+**Returns:**
+
+- [(string)](https://cplusplus.com/reference/string/string/): Result of the executed command.
+
+```cpp {class="line-numbers linkable-line-numbers"}
+result_map = client.do_command(command);
+```
```

Comment:
I was not able to find a good example for a client program using `do_command()` in CPP (examples I found are all of modules re-implementing, not client code using). Apologies if I missed it. This guess here of mine is likely quite wrong.

Reply:
@benjirewis Thanks so much! I think I'll take the opportunity to add this to the generic component page too, which previously did not include C++ guidance. Would the following be similarly accurate for the generic component? (only changed the first line):

```
auto my_generic = robot->resource_by_name<GenericComponent>("my_generic_component");
auto example = std::make_shared<ProtoType>(std::string("example"));
AttributeMap command =
    std::make_shared<std::unordered_map<std::string, std::shared_ptr<ProtoType>>>();
command->insert({{std::string("command"), example}});
auto resp = my_generic->do_command(command);
```

Thanks!

Reply:
Code looks good in both places apart from saying that `command` and returned value from `do_command` are `string`s. They are `AttributeMap`s.